Sources/FirebaseCore/FutureProtocol.swift

```diff
@@ -47,10 +47,10 @@ public extension FutureProtocol {
     return String(cString: errorMessageUnsafe)
   }
 
-  var resultAndError: (ResultType?, Error?) {
+  func resultAndError<ErrorType>(constructError: ((Int32, String)) -> ErrorType) -> (ResultType?, ErrorType?) {
```

Comment:
Since you are intending to use this as a trailing closure, I think that you should name the label `_` (trailing closures are generally with unnamed parameters). For the interior label, I think that `error` is a better name.

Reply:
Huh, it seems fairly conventional for these to be named. I can label it `_` if you prefer. Why do you prefer `error` over `constructError`?

Reply:
Renaming this to `error` creates compilation errors as `error` is already used as a local variable (and is also a member function). I will just stick with `constructError` to avoid those issues.

Reply:
Ah, okay, but please do label it as `_`. I prefer the explicit `_` to indicate that the label may be omitted - which is the case with trailing closures, unless you intend this to be used as:

```swift
resultAndError constructError: { ... }
```

The label being `constructError` seems ... unnecessarily verbose is all.